tests/validators/regex_validator_test.py

```diff
@@ -161,6 +161,86 @@ def test_invalid_regex_pattern():
             RegexValidator('[')
         assert str(exception_info.value) == 'unterminated character set at position 0'
 
+    # Test RegexValidator with output template
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'regex_pattern, output_template, input_data, expected_output', [
+            # Empty template returns the original string
+            (r'.*', r'', 'some input', 'some input'),
+
+            # Constant/no match template
+            (r'.*', r'constant template', 'lorem ipsum lololol', 'constant template'),
+
+            # With numeric groups for matches
+            (r'(\w+) (\w+)', r'First: \1, Last: \2', "Isaac Newton", "First: Isaac, Last: Newton"),
+
+            # With named groups for matches
+            (r'(?P<first>\w+) (?P<last>\w+)', r'First: \g<first>, Last: \g<last>', "Isaac Newton", "First: Isaac, Last: Newton"),
+        ]
+    )
+    def test_templated_output_valid(regex_pattern, output_template, input_data, expected_output):
+        """ Check that RegexValidator give templated outputs correctly. """
+        validator = RegexValidator(regex_pattern, output_template)
+        assert validator.validate(input_data) == expected_output
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'regex_pattern, output_template, input_data, exception_message', [
+            # No numeric match group
+            (
+                r'.*',
+                r'\1',
+                'lmao',
+                "invalid group reference 1 at position 1",
+            ),
+
+            # Nonexisting numeric match group
+            (
+                r'(\w+) (\w+)',
+                r'\3',
+                "Isaac Newton",
+                "invalid group reference 3 at position 1",
+            ),
+
+            # Nonexisting named match group
+            (
+                r'(?P<first>\w+) (\w+)',
+                r'\g<last>',
+                "Isaac Newton",
+                "unknown group name 'last'",
+            ),
+        ]
+    )
+    def test_templated_output_invalid(regex_pattern, output_template, input_data, exception_message):
+        """ Check that RegexValidator raises exception for invalid templates. """
+        validator = RegexValidator(regex_pattern, output_template)
+        with pytest.raises((re.error, IndexError)) as exception_info:
+            validator.validate(input_data)
+        assert str(exception_info.value) == exception_message
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'input_data', [
+            'name.jpgg',
+            'nonamepng',
+            '.png',
+            'noext.',
+        ]
+    )
+    def test_input_invalid(input_data):
```

Comment:
Just a small suggestion for the test names to keep them consistent with the rest:
`test_templated_output_invalid` would be a better name for this test (test the validator with invalid input). The test above could be called `test_invalid_output_template` or `test_malformed_output_template` then. :)

Reply:
OK. Thanks for the help. I struggle with coming up with accurate/consistent names for the tests.